internal/handler/msg_find.go

```diff
@@ -102,10 +104,19 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 	}
 
 	cancel := func() {}
+	var findDone atomic.Bool
+
 	if params.MaxTimeMS != 0 {
-		// It is not clear if maxTimeMS affects only find, or both find and getMore (as the current code does).
-		// TODO https://github.com/FerretDB/FerretDB/issues/2984
-		ctx, cancel = context.WithTimeout(ctx, time.Duration(params.MaxTimeMS)*time.Millisecond)
+		ctx, cancel = context.WithCancel(ctx)
+		go func() {
```

Comment:
Hmm, this gorutine will run for the whole duration of `MaxTimeMS`, even if `findDone.Load()` was true a while ago. 

Wouldn't something like this work?

```go
for {
		select {
		case <-ticker.C:
			canel()
		case <-signalChan:
			return
		}
	}
```

and `signalChan` is closed the same way as `findDone` is set as true.

Reply:
I thought about something like that, though at this point it felt that introducing another go routine instead of ticker might be more suitable (as we only need one signal and not entire ticker). Although I remember that I've implemented something similar in some previous PR, and there were some objections in terms of complexity because of goroutines.

But yeah, in this case it seems to be more reasonable so will change that quickly and let's wait for the comments:)